notebook/static/base/js/utils.js

```diff
@@ -772,7 +772,12 @@ define([
     };
     
     var ajax = function (url, settings) {
-        // like $.ajax, but ensure Authorization header is set
+        // like $.ajax, but ensure XSRF or Authorization header is set
+        if (typeof url === "object") {
+            // called with single argument: $.ajax({url: '...'})
+            settings = url;
+            url = settings.url;
```

Comment:
Don't you think it is necessary to remove `settings.url` from `settings` after defining the `url` here? I guess it is not harmful but I think it would complete the "split" of the two objects.
In any case, just a minor comment, LGTM as is as well.